SafetyFirst/Set.fs

```diff
@@ -0,0 +1,220 @@
+module SafetyFirst.Set
+
+module NonEmpty = 
+
+  /// <summary>
+  /// Creates a new NonEmptySet with the provided head and tail.  
+  /// The tail is constrained to be finite.  If the tail is infinite,
+  /// use Seq.NonEmpty.create instead.
+  /// </summary>
+  let create head tail : NonEmptySet<_> = NonEmpty (set (Seq.append [head] tail)) 
+
+  /// <summary>
+  /// Returns a set that contains one item only.
+  /// </summary>
+  let singleton head : NonEmptySet<_> = NonEmpty (Set.singleton head)
+
+  /// <summary>
+  /// Returns the lowest element in the set according to the ordering being used for the set.
+  /// </summary>
+  let minElement (NonEmpty xs : NonEmptySet<_>) = Set.minElement xs
+
+  /// <summary>
+  /// Returns the highest element in the set according to the ordering being used for the set.
+  /// </summary>
+  let maxElement (NonEmpty xs : NonEmptySet<_>) = Set.maxElement xs
+
+  /// <summary>
+  /// Returns a new set with an element added to the set.
+  /// If you're looking for an overload where the input set can be empty,
+  /// use <c>Set.create</c> instead. 
+  /// No exception is raised if the set already contains the given element.
+  /// </summary>
+  let add x (NonEmpty xs : NonEmptySet<_>) : NonEmptySet<_> = NonEmpty (Set.add x xs)
+
+  /// <summary>
+  /// Evaluates to "true" if the given element is in the given set.
+  /// </summary>
+  let contains x (NonEmpty xs : NonEmptySet<_>) = Set.contains x xs
+
+  /// <summary>
+  /// Returns the number of elements in the set. Same as size.
+  /// </summary>
+  let count (NonEmpty xs : NonEmptySet<_>) = Set.count xs
+
+  // should xs be NonEmpty?  ys?  Both?  Should we have overloads for all 3?
+  // let difference xs ys = Set.difference xs ys
+
+  /// <summary>
+  /// Tests if any element of the collection satisfies the given predicate. 
+  /// If the input function is <c>p</c> and the elements are <c>i0...iN</c> 
+  /// then computes <c>p i0 or ... or p iN</c>.
+  /// </summary>
+  let exists predicate (NonEmpty xs : NonEmptySet<_>) = Set.exists predicate xs
+
+  /// <summary>
+  /// Returns a new collection containing only the elements of the collection 
+  /// for which the given predicate returns True.
+  /// </summary>
+  let filter predicate (NonEmpty xs : NonEmptySet<_>) = Set.filter predicate xs
+
+  /// <summary>
+  /// Applies the given accumulating function to all the elements of the set.
+  /// </summary>
+  let fold folder initialState (NonEmpty xs : NonEmptySet<_>) = Set.fold folder initialState xs
+
+  /// <summary>
+  /// Applies the given accumulating function to all the elements of the set.

```

Comment:
really, is this supposed to have the same comment as regular `fold`?

Reply:
The regular `Set.foldBack` function has the same comment as `Set.fold`.  I've noticed that the documentation of the `Set` module in general is a little weak.  Since the `fold` and `foldBack` functions here effectively do the same thing as `Seq.fold` and `Seq.foldBack`, I'll just copy the documentation for those functions instead.

Reply:
✔️